src/feature-lifecycle/stabilization.md

```diff
@@ -25,3 +40,71 @@ Const functions can be stabilized in a PR that replaces `#[rustc_const_unstable]
 
 Where `unsafe` and `const` is involved, e.g., for operations which are "unconst", that the const safety argument for the usage also be documented. That is, a `const fn` has additional determinism (e.g. run-time/compile-time results must correspond and the function's output only depends on its inputs...) restrictions that must be preserved, and those should be argued when `unsafe` is used.
+
+## Stabilization PR for Library Features
+
+Once we have decided to stabilize a feature, we need to have a PR that actually makes that stabilization happen. These kinds of PRs are a great way to get involved in Rust, as they're typically small -- just updating attributes.
+
+Here is a general guide to how to stabilize a feature -- every feature is different, of course, so some features may require steps beyond what this guide talks about.
+
+### Update the stability attributes on the items
+
+Library items are marked unstable via the `#[unstable]` attribute, like this:
+
+```rust,ignore
+#[unstable(feature = "total_cmp", issue = "72599")]
+pub fn total_cmp(&self, other: &Self) -> crate::cmp::Ordering { ... }
+```
+
+You'll need to change that to a `#[stable]` attribute with a version:
+
+```rust,ignore
+#[stable(feature = "total_cmp", since = "1.61.0")]
+```
+
+Note that, the version number is updated to be the version number of the stable release where this feature will appear. This can be found by consulting [the forge](https://forge.rust-lang.org/#current-release-versions). Specifically, you'll want to use the version labelled "Nightly". That's two versions higher than the current stable release, as what's currently in beta will be the next stable release, and any change you're making now will be in the one after that.
+
+### Remove feature gates from doctests
+
+All the doctests on the items being stabilized will be enabling the unstable feature, so now that it's stable those attributes are no longer needed and should be removed.
+
+`````diff
+ /// # Examples
+ ///
+ /// ```
+-/// #![feature(total_cmp)]
+-///
+ /// assert_eq!(0.0_f32.total_cmp(&-0.0), std::cmp::Ordering::Greater);
+ /// ```
+`````
+
+The most obvious place to find these is on the item itself, but it's worth searching the whole library.  Often you'll find other unstable methods that were also using it in their tests.
+
+### Remove feature gates from the compiler
+
+The compiler builds with nightly features allowed, so you may find uses of the feature there as well.  These also need to be removed.
+
+```diff
+ #![feature(once_cell)]
+ #![feature(never_type)]
+-#![feature(total_cmp)]
+ #![feature(trusted_step)]
+ #![feature(try_blocks)]
+```
+
+## Stabilization PR Checklist
+
+To stabilize a feature, follow these steps:
+
+0. (Optional) For partial stabilizations, create a new tracking issue for either the subset being stabilized or the subset being left unstable, whichever makes the most sense based on the situation.
```

Comment:
Pondering: It feels odd to make a new tracking issue part of the partial *stabilization*, since it'll just be closed immediately if that subset does actually stabilize.

This is one place where I think the "make a stabilization PR" flow is actually better.  It's *easier* to make a stabilization PR than a "split up the feature gate" PR.  And it's less impactful to nightly users -- if it's split, they all have to add more feature gates, which is particularly obnoxious if the goal is just to stabilize the new one ASAP, after which they'd need to remove it again.

Reply:
Agreed. In my experience asking on Zulip or the existing tracking issue is sufficient to gauge interest in a partial stabilization, which ensures time isn't wasted on PRs. Creating a new tracking issue seems noisy and somewhat disruptive to nightly users.

Reply:
We wanted to avoid suggesting Zulip here to avoid splitting discussions across multiple platforms, but yea generally our goal here is to communicate how to move these issues forward most effectively / least disruptively. I'll update this section to clarify that for partial stabilizations they can skip creating a new tracking issue and do the stabilization report as part of the PR doing said stabilization.

Reply:
Sounds good to me. Ideally we'd be able to have multiple FCPs per issue, but I know that's not currently possible.

Reply:
I actually like the different items for different FCPs, myself.  A PR for the stabilization make it much clearer exactly what an FCP is stabilizing, if it's not "everything with this feature".  And GitHub is not great at hundreds-of-posts anyway, which can easily happen with a bunch of FCPs in the same item.